packages/cli/src/daemon-config.ts

```diff
@@ -88,18 +88,7 @@ export class DaemonHTTPApiConfig {
   /**
    * Port to listen on.
    */
-  @jsonMember(AnyT, {
-    serializer: (inPort) => {
-      const validPort = Number(inPort)
-      if (inPort == null) {
-        return inPort
-      } else if (isNaN(validPort) || validPort > 65535) {
-        console.error('Invalid port number passed.')
-        process.exit(1)
-      }
-      return validPort
-    },
-  })
+  @jsonMember(AnyT)
```

Comment:
Number
```suggestion
  @jsonMember(Number)
```

Reply:
Need to verify that this indeed works to switch it back into `Number`. I had an issue with that in the past because of the instantiation with the value in memory (hence why I moved the coercion initially into the member serializer).

Reply:
oh right, that was where the original error we were seeing came from, the whole reason we started this line of work.  If using a serializer here stopped that error from happening, then it seems like the serializer was indeed being used?

Reply:
yeah, going full circle with this one. It was being used but in a "hacky" form (using serializer vs. deserializer etc.) that didn't fully work for the config  so pushing it into the `CeramicDaemon` class addresses now all use cases and ensures that either the port is correctly parsed or the daemon exits.

Reply:
I'm so confused.  How could it work but hacky?  When a config file is loaded and parsed, either it calls the serializer or it calls the deserializer.  It doesn't call both.  Or am I just totally misunderstanding how the option parser works?

Reply:
Added comment as per our discussion to explain the type requirement. Once you give your final OK going to merge it in.